pkg/frontend/querymiddleware/astmapper/instant_splitting_stats.go

```diff
@@ -19,3 +22,39 @@ func (s *InstantSplitterStats) AddSplitQueries(num int) {
 func (s *InstantSplitterStats) GetSplitQueries() int {
 	return s.splitQueries
 }
+
+// SetNoOpSmallRangeInterval set query as a no operation due to a small range interval.
+func (s *InstantSplitterStats) SetNoOpSmallRangeInterval() {
```

Comment:
I would keep easier and just have 1 setter which takes in input the reason. We only track the 1st reason encountered. If a query is not splittable both because has a subquery AND because of the time interval, than I'm sorry, we'll just track one of the 2 (the one we encounter first). We just want to have a rough idea, not a super accurate measurement.

Reply:
Ok, good idea. Thanks ⭐ 
Addressed in https://github.com/grafana/mimir/pull/2642/commits/11d91d816a1e5b36ec02988e57d151e4865df9ae